src/main/groovy/cruise/umple/UmplePlugin.groovy

```diff
@@ -5,16 +5,29 @@ import cruise.umple.UmpleConsoleConfig
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 
-class UmplePlugin implements Plugin<Project> {
+class UmpleGradlePlugin implements Plugin<Project> {
 
     private UmpleConsoleConfig cfg
 
     @Override
     void apply(final Project project) {
 
-
-        project.task('hello') << {
-            println "Hello, World!"
+        project.task('compileUmpleFile') << {
+			UmpleConsoleMain(cfg)
+		
+			if(project.hasProperty("umpleArgs"))
+			{
+				// arguments are specified through gradle by -P, separated by commas (-P is for project properties)
+				// eg: "gradle compileUmpleFile 'P-umpleArgs=test.ump,-g,Java'"
+				args(umpleArgs.split(','))
```

Comment:
I just saw your comment on umple/umple#751, this would address your remark about argument parsing. 

If you specify an argument with `-P`, it'll show up as a property so you can check with the same as [this](https://github.com/umple/umple.gradle/pull/2/files/2b36cbc04dd11a911c20089e10ee8e5fba0344a9#diff-a2ea8ddc483a60a14b18aca63ef4b80fR18). Should solve that problem!